app-backend/tool/src/main/scala/eval/Interpreter.scala

```diff
@@ -51,6 +52,49 @@ object Interpreter extends LazyLogging {
       eval(args.head).classify(breaks.toBreakMap)
   }
 
+  def overrideParams(
+    ast: MapAlgebraAST,
+    overrides: Map[UUID, ParamOverride]
+  ): Interpreted[MapAlgebraAST] = ast match {
+    /* Can't override data Sources */
+    case s: Source => Valid(s)
+
+    /* Nodes which can be overridden */
+    case c: Constant => overrides.get(c.id) match {
+      case None => Valid(c)
+      case Some(ParamOverride.Constant(const)) => Valid(c.copy(constant = const))
+      case Some(_) => Invalid(NonEmptyList.of(InvalidOverride(c.id)))
+    }
```

Comment:
I worry about adding this logic to every tile request - does this seem like premature optimization to you?